stdlib/public/core/StringBridge.swift

```diff
@@ -537,6 +631,200 @@ internal func _NSStringFromUTF8(_ s: UnsafePointer<UInt8>, _ len: Int)
   )._bridgeToObjectiveCImpl()
 }
 
+@_fixed_layout
+@usableFromInline
+@objc internal final class _SwiftNSMutableString :
+  _SwiftNativeNSMutableString, _AbstractStringStorage {
+  private var _contents: String
+  
+  internal init(_ str: String) {
+    _contents = str
+    super.init()
+  }
+  
+  final internal var asString: String {
+    @_effects(readonly) @inline(__always) get {
+      return _contents
+    }
+  }
+  
+  @inline(__always)
+  final func withFastUTF8<R>(
+    _ f: (UnsafeBufferPointer<UInt8>) throws -> R
+  ) rethrows -> R {
+    return try _contents._guts.withFastUTF8(f)
+  }
+  
+  @inline(__always)
+  internal var start: UnsafePointer<UInt8>? {
+     let guts = _contents._guts
+     if !guts.isSmall && guts.isFastUTF8 {
+       return guts._object.fastUTF8.baseAddress!
+     }
+     return nil
+  }
+  
+  @inline(__always)
+  internal var count: Int { return _contents.utf8.count }
+  
+  @inline(__always)
+  final internal var isASCII: Bool {
+    return _contents._guts.isASCII
+  }
+  
+  @objc(length)
+  final internal var UTF16Length: Int {
+    @_effects(readonly) get {
+      return _contents.utf16.count // UTF16View special-cases ASCII for us.
+    }
+  }
+
+  @objc
+  final internal var hash: UInt {
+    @_effects(readonly) get {
+      if isASCII {
+        return _contents._guts.withFastUTF8 {
+          return _cocoaHashASCIIBytes($0.baseAddress!, length: $0.count)
+        }
+      }
+      return _cocoaHashString(self)
+    }
+  }
+
+  @objc(characterAtIndex:)
+  @_effects(readonly)
+  final internal func character(at offset: Int) -> UInt16 {
+    return _contents.utf16[_contents._toUTF16Index(offset)]
+  }
+
+  @objc(getCharacters:range:)
+  @_effects(releasenone)
+  final internal func getCharacters(
+    _ buffer: UnsafeMutablePointer<UInt16>, range aRange: _SwiftNSRange
+  ) {
+    _getCharacters(buffer, aRange)
+  }
+
+  @objc
+  final internal var fastestEncoding: UInt {
+    @_effects(readonly) get {
+      if isASCII {
+        return _cocoaASCIIEncoding
+      }
+      return _cocoaUTF8Encoding
+    }
+  }
+
+  @objc(_fastCStringContents:)
+  @_effects(readonly)
+  final internal func _fastCStringContents(
+    _ requiresNulTermination: Int8
+  ) -> UnsafePointer<CChar>? {
+    let guts = _contents._guts
+    if !guts.isSmall && guts.isFastUTF8 && guts.isASCII {
+      return guts._object.fastUTF8.baseAddress!._asCChar
+    }
+    return nil
+  }
+
+  @objc(UTF8String)
+  @_effects(readonly)
+  final internal func _utf8String() -> UnsafePointer<UInt8>? {
+    let guts = _contents._guts
+    if !guts.isSmall && guts.isFastUTF8 {
+      return guts._object.fastUTF8.baseAddress!
+    }
+    return nil
```

Comment:
[This one's not allowed to return `nil`.](https://developer.apple.com/documentation/foundation/nsstring/1411189-utf8string)

Reply:
ah, good catch. I should… go review the other overrides of that and make sure they're not wrong O_O